src/components/modal.js

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable import/no-cycle */
+import { addNewComment, getPopUpCharacterComments } from '../apiCalls';
+import commentsCounter from '../commentsCounter';
+
+export const displayComments = async (characterId) => {
+  await getPopUpCharacterComments(characterId).then((comments) => {
+    const commentsCount = commentsCounter(comments);
+    document.getElementById('commentCount').innerText = commentsCount
+      ? `Comments: ${commentsCount}`
+      : `Comments: ${0}`;
+    const modalComments = document.querySelector('.modal-comments');
+    modalComments.innerHTML = '';
+    comments.forEach((comment) => {
+      modalComments.innerHTML += `
+          <p class="modal-comment">${comment.creation_date} ${comment.username}: ${comment.comment}</p>
+          `;
+    });
+  });
+};
+
+const modal = async (character, characterId) => {
+  const popUpModal = document.createElement('div');
+  popUpModal.classList = 'modal';
+  popUpModal.innerHTML = `
+    <div class="modal-container">
+      <button class='close-modal'>X</button>
+      <div class="modal-img">
+        <img src=${character.image} alt="modal-img">
+        <h2>${character.name}</h2>
+      </div>
+      <div class="modal-info-container">
+        <div class="modal-info">
+          <p class="modal-info-text">
+            House: ${character.house}
+          </p>
+          <p class="modal-info-text">
+            Gender: ${character.gender}
+          </p>
+          <p class="modal-info-text">
+            Ancestry: ${character.ancestry}
+          </p>
+          <p class="modal-info-text">
+            Species: ${character.species}
+          </p>
+        </div>
+      </div>
+
+      ${displayComments(characterId)}
+    
```

Comment:
- [OPTIONAL] It will be good to move this out of the HTML div to be called outside the innerHTML as it is currently being displayed as `[object Promise]` in the popup modal.

Reply:
@ZeenatLawal thanks for the suggestion. I will implement it